cirq-core/cirq/ops/uniform_superposition_gate.py

```diff
@@ -0,0 +1,127 @@
+
+    References:
+        [SV24]
+            A. Shukla and P. Vedula, "An efficient quantum algorithm for preparation of uniform
+            quantum superposition states," Quantum Information Processing, 23(38): pp. 1-32 (2024).
+    """
+
+    def __init__(self, m_value: int, num_qubits: int) -> None:
+        """Initializes UniformSuperpositionGate.
+
+        Args:
+            m_value (int): The number of computational basis states with amplitude 1/sqrt(M).
+            num_qubits (int): The number of qubits used.
+
+        Raises:
+            ValueError: If `m_value` is not a positive integer, or
+                if `num_qubits` is not an integer greater than or equal to log2(m_value).
+        """
+        if not (isinstance(m_value, int) and (m_value > 0)):
+            raise ValueError('m_value must be a positive integer.')
+        if not (isinstance(num_qubits, int) and (num_qubits >= math.log2(m_value))):
+            raise ValueError(
+                'num_qubits must be an integer greater than or equal to log2(m_value).'
+            )
+        self._m_value = m_value
+        self._num_qubits = num_qubits
+
+    def _decompose_(self, qubits: Sequence['cirq.Qid']) -> 'cirq.OP_TREE':
+        """Decomposes the gate into a sequence of standard gates.
+
+        Args:
+            qubits (list[cirq.Qid]): Qubits to apply the gate on.
+
+        Yields:
+            cirq.Operation: Operations implementing the gate.
+        """
+        qreg = list(qubits)
+        qreg.reverse()
+
+        if self._m_value == 1:  #  if m_value is 1, do nothing
+            return
+        if (self._m_value & (self._m_value - 1)) == 0:  # if m_value is an integer power of 2
+            m = self._m_value.bit_length() - 1
+            yield H.on_each(qreg[:m])
+            return
+        k = self._m_value.bit_length()
+        l_value = []
+        for i in range(self._m_value.bit_length()):
+            if (self._m_value >> i) & 1:
+                l_value.append(i)  # Locations of '1's
+
+        yield X.on_each(qreg[q_bit] for q_bit in l_value[1:k])
+        m_current = 2 ** (l_value[0])
+        theta = -2 * np.arccos(np.sqrt(m_current / self._m_value))
+        if l_value[0] > 0:  # if m_value is even
+            yield H.on_each(qreg[: l_value[0]])
+
+        yield ry(theta).on(qreg[l_value[1]])
+
+        for i in range(l_value[0], l_value[1]):
+            yield H(qreg[i]).controlled_by(qreg[l_value[1]], control_values=[False])
+
+        for m in range(1, len(l_value) - 1):
+            theta = -2 * np.arccos(np.sqrt(2 ** l_value[m] / (self._m_value - m_current)))
+            yield controlled_gate.ControlledGate(ry(theta), control_values=[False])(
+                qreg[l_value[m]], qreg[l_value[m + 1]]
+            )
+            for i in range(l_value[m], l_value[m + 1]):
+                yield controlled_gate.ControlledGate(H, control_values=[False])(
+                    qreg[l_value[m + 1]], qreg[i]
+                )
```

Comment:
```suggestion
                yield H.on(qreg[i]).controlled_by(qreg[l_value[m + 1]], control_values=[0])
```

Reply:
@anurudhp Thanks for your input. Suggested changes were made